### PR TITLE
devices: debian: set_link_status() - To set the link status

### PR DESCRIPTION
**Pull request by sravichandran-contractor**
devices: debian_wifi: - Function to connect wifi
    wlan_ssid_connect() - connect to wifi in security disabled mode
    wlan_ssid_disconnect() - disconnect wifi in security disabled mode
    start_lan_client() - Function to call the DHCP process
    linux_wifi_client() - generic function which scan ssid and verifies wifi connectivity

lib: common: Functions for file open, snmp set and get
    file_open() - function to open a file and read the values
    snmp_mib_set() - SNMP set function to set values
    snmp_mib_get() - SNMP get function to get values

lib: wifi: Generic snmp set and get functions for wifi
    wifi_stub() - Generic class for wifi functions

### CODE REVIEW

File: devices/debian_wifi.py [hunk 1]
```diff
@@ -53,3 +54,38 @@ def wlan_connectivity(self):
...
+    def wlan_ssid_connect(self, ssid_name):
+        self.sendline("iwconfig %s essid %s" % (self.iface_wlan,ssid_name))
+        self.expect(self.prompt)
+
+    def wlan_ssid_disconnect(self):
+        self.sendline("iw dev %s disconnect" % self.iface_wlan)
+        self.expect(self.prompt)
+
+    def start_lan_client(self, wan_gw=None):
+        self.iface_dut = self.iface_wlan
```
Comment (mattsm) [inline devices/debian_wifi.py]: Do we need iface_wlan at all? Should just use iface_dut instead?

File: devices/debian.py [hunk 2]
```diff
@@ -966,6 +966,11 @@ def link_up(self, interface):
         else:
             return None
 
+    def set_link_status(self, interface, status):
```
Comment (mattsm) [inline devices/debian.py]: Can we rename this to set_link_state? This is just turning the interface on and off, right?

File: tests/lib/wifi.py [hunk 3]
```diff
@@ -177,3 +178,165 @@ def uciSetWifiSecurity(board, vap_iface, security):
...
+    def set_password_verify(self, *args, **kwargs):
+        raise Exception("Not implemented!")
+    def enable_channel_util_verify(self, *args, **kwargs):
+        raise Exception("Not implemented!")
+    def set_operating_mode_verify(self, *args, **kwargs):
+        raise Exception("Not implemented!")
+    def set_bandwidth_verify(self, *args, **kwargs):
+        raise Exception("Not implemented!")
+    def set_channel_no_verify(self, *args, **kwargs):
+        raise Exception("Not implemented!")
```
Comment (mattsm) [inline tests/lib/wifi.py]: This class looks ok- but why is everything appended with _verify?

File: tests/lib/wifi.py [hunk 4]
```diff
@@ -177,3 +178,165 @@ def uciSetWifiSecurity(board, vap_iface, security):
...
+       return mib_out
+    else:
+       return None
+
+def get_broadcast(device, board, iface_ip, mib_name, index):
+    mib_out = snmp_mib_get(device,board,iface_ip,mib_name,index)
+    if mib_out != None:
+        return mib_out
+    else:
+        return None
```
Comment (mattsm) [inline tests/lib/wifi.py]: I don't think we need these functions. I think the device class handler for wifi_snmp() can just call these. These might not be generic enough for other board types anyways so we will just let the device classes handle it all.

File: tests/lib/common.py [hunk 5]
```diff
@@ -389,3 +392,64 @@ def configure_postfix(device):
...
+            if "json" in extension:
+                read_data = json.load(f)
+            elif "txt" in extension:
+                read_data = f.read()
+            return read_data
+        except Exception as e:
+            print "Could not read file from the path:", file_path
+            raise (e)
+
+def snmp_mib_set(device, board, iface_ip, mib_name, index, set_type, set_value, timeout=10, retry=3):
```
Comment (mattsm) [inline tests/lib/common.py]: These SNMP functions are OK- but they really could be in their own commit so we could let them through and not be held up by the other reviews.

File: tests/lib/common.py [hunk 6]
```diff
@@ -389,3 +392,64 @@ def configure_postfix(device):
...
+    with open(file_path) as f:
+        try:
+            if "json" in extension:
+                read_data = json.load(f)
+            elif "txt" in extension:
+                read_data = f.read()
+            return read_data
+        except Exception as e:
+            print "Could not read file from the path:", file_path
+            raise (e)
```
Comment (mattsm) [inline tests/lib/common.py]: Can we keep this a bit simplier for now:
```
def file_open_json(file):
    with open(file) as f:
        return json.load(f)
```

We should throw an exception if something fails, since we're only loading a json file.

Also, this could and should be in it's own commit.

File: devices/debian.py (hunk 2 above)
Comment (sravichandran-contractor) [reply devices/debian.py]: Sure will rename the same

File: tests/lib/wifi.py (hunk 3 above)
Comment (sravichandran-contractor) [reply tests/lib/wifi.py]: Since the function in device class does a set and verify of the values we appended with _verify. The same name has been used here. Should we remove this?

File: tests/lib/common.py (hunk 6 above)
Comment (sravichandran-contractor) [reply tests/lib/common.py]: Agreed will change the same

File: devices/debian_wifi.py (hunk 1 above)
Comment (sravichandran-contractor) [reply devices/debian_wifi.py]: If I use iface_dut it is taking eth0 as default. As I wasn't the interface to be wlan1 am passing the iface_wlan1 here

File: tests/lib/wifi.py (hunk 4 above)
Comment (sravichandran-contractor) [reply tests/lib/wifi.py]: Sure. Will dissolve this function and will call snmp set and get directly in the device class handler for wifi_snmp()

**Comment by mattsm** [comment]
Merged manually
